cli/src/main/java/org/infinispan/cli/commands/Credentials.java

```diff
@@ -0,0 +1,205 @@
+   public boolean isHelp() {
+      return help;
+   }
+
+   @Override
+   public CommandResult exec(ContextAwareCommandInvocation invocation) {
+      // This command serves only to wrap the sub-commands
+      invocation.println(invocation.getHelpInfo());
+      return CommandResult.FAILURE;
+   }
+
+   static String getPasswordInteractively(ContextAwareCommandInvocation invocation, String prompt, String confirmPrompt) throws InterruptedException {
+      String password = null;
+      while (password == null || password.isEmpty()) {
+         password = invocation.getShell().readLine(new Prompt(prompt, '*'));
+      }
+      if (confirmPrompt != null) {
+         String confirm = null;
+         while (confirm == null || !confirm.equals(password)) {
+            confirm = invocation.getShell().readLine(new Prompt(confirmPrompt, '*'));
+         }
+      }
+      return password;
+   }
+
+   static KeyStoreCredentialStore getKeyStoreCredentialStore(String path, String type, boolean create, char[] password) throws CredentialStoreException {
+      KeyStoreCredentialStore store = new KeyStoreCredentialStore();
+      final Map<String, String> map = new HashMap<>();
+      map.put("location", path);
+      map.put("keyStoreType", type);
+      map.put("create", Boolean.toString(create));
+      store.initialize(
+            map,
+            new CredentialStore.CredentialSourceProtectionParameter(
+                  IdentityCredentials.NONE.withCredential(new PasswordCredential(ClearPassword.createRaw(ClearPassword.ALGORITHM_CLEAR, password)))),
+            null
+      );
+      return store;
+   }
+
+   @CommandDefinition(name = Add.CMD, description = "Adds a credential to a key store")
+   public static class Add extends CliCommand {
+      public static final String CMD = "add";
+
+      @Argument(description = "The name (alias) of the credential", required = true)
+      String alias;
+
+      @Option(description = "The path to a keystore. A new one will be created if it doesn't exist.", completer = FileOptionCompleter.class, defaultValue = CREDENTIALS_PATH)
+      Resource path;
+
+      @Option(description = "The password to protect the credential store", shortName = 'p')
+      String password;
+
+      @Option(description = "The type of the credential store", shortName = 't', defaultValue = STORE_TYPE)
+      String type;
+
+      @Option(description = "The credential to store", shortName = 'c')
+      String credential;
+
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) throws CommandException {
+         try {
+            if (password == null) {
+               password = getPasswordInteractively(invocation, MSG.credentialToolPassword(), path.exists() ? null : MSG.credentialToolPasswordConfirm());
```

Comment:
nitpick: the prompt should be different if the file exists.
`Set a password for the credential store` => if the file doesn't exist.
something like `Credential store password` => if the file exists.

Reply:
ps. use the second prompt for remove/ls commands.